lib/src/client_browser.dart

```diff
@@ -156,7 +156,7 @@ class ClientBrowser extends ClientBase with ClientMixin {
 
     while (offset < size) {
       var chunk;
-      final end = min(offset + CHUNK_SIZE - 1, size - 1);
+      final end = min(offset + CHUNK_SIZE, size);
```

Comment:
@lohanidamodar do we really want to make this change?

Reply:
Yes